### PR TITLE
Use mTLS for ServiceMonitor

### PR DESCRIPTION
**Pull request by jeremyharisch**
**Description**
Regarding [this issue](https://github.com/kyma-project/kyma/issues/11179) it is now possible to scrape metrics using mTLS. This PR should be merged after [this PR](https://github.com/kyma-project/kyma/pull/11994), but before [this PR](https://github.com/kyma-project/kyma/pull/12013).

Changes proposed in this pull request:

- Added mTls configuration to ServiceMonitor
- Added new `peerAuthentication` policy to make use of mTls

**Related issue(s)**

[#11179](https://github.com/kyma-project/kyma/issues/11179)

### CODE REVIEW

File: monitoring-custom-metrics/deployment/policy.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: security.istio.io/v1beta1
```
Comment (a-thaler) [inline monitoring-custom-metrics/deployment/policy.yaml:1]: Why the resource was not needed before? Why it is needed now?

Comment (jeremyharisch) [reply monitoring-custom-metrics/deployment/policy.yaml:1]: The PeerAuthentication is needed to enabled Prometheus to scrape using mTLS. And if you look into the docs of PeerAuthentication from [Istio](https://istio.io/latest/docs/reference/config/security/peer_authentication/), it is the way how it should be defined.

File: monitoring-custom-metrics/deployment/service-monitor.yaml [hunk 2]
```diff
@@ -16,5 +16,11 @@ spec:
   endpoints:
     - port: web
       interval: 10s
+      scheme: https
+      tlsConfig: 
+        caFile: /etc/prometheus/secrets/istio.default/root-cert.pem
+        certFile: /etc/prometheus/secrets/istio.default/cert-chain.pem
+        keyFile: /etc/prometheus/secrets/istio.default/key.pem
+        insecureSkipVerify: true  # Prometheus does not support Istio security naming, thus skip verifying target pod ceritifcate
```
Comment (alexandra-simeonova) [inline monitoring-custom-metrics/deployment/service-monitor.yaml:24]:
```suggestion
        insecureSkipVerify: true  # Prometheus does not support Istio security naming, thus skip verifying target Pod certificate
```

**Review by alexandra-simeonova — DISMISSED** [review]

File: monitoring-custom-metrics/deployment/deployment.yaml [hunk 3]
```diff
@@ -35,29 +37,13 @@ apiVersion: v1
...
-    port: 8080
-    protocol: TCP
-
----
-kind: Service
-apiVersion: v1
-metadata:
-  name: sample-metrics-8081
-  labels:
     k8s-app: metrics
```
Comment (a-thaler) [inline monitoring-custom-metrics/deployment/deployment.yaml:41]: I think k8s-app is a bad label key. Can we switch to "app"?

File: monitoring-custom-metrics/deployment/service-monitor.yaml [hunk 4]
```diff
@@ -16,5 +16,11 @@ spec:
   endpoints:
```
Comment (a-thaler) [inline monitoring-custom-metrics/deployment/service-monitor.yaml:16]: The targetLabels section above should not be required, please double-check and remove

File: monitoring-custom-metrics/deployment/service-monitor.yaml [hunk 5]
```diff
@@ -16,5 +16,11 @@ spec:
...
     - port: web
```
Comment (a-thaler) [inline monitoring-custom-metrics/deployment/service-monitor.yaml:17]: you renamed the port, have you tested the setup?

Comment (jeremyharisch) [reply monitoring-custom-metrics/deployment/service-monitor.yaml:17]: Tested it now, was indeed an error

File: monitoring-custom-metrics/deployment/service-monitor.yaml (hunk 4 above)
Comment (jeremyharisch) [reply monitoring-custom-metrics/deployment/service-monitor.yaml:16]: Not required

**Review by a-thaler — APPROVED** [review]

**Review by NHingerl — DISMISSED** [review]

**Review by a-thaler — DISMISSED** [review]

**Review by NHingerl — APPROVED** [review]

**Review by a-thaler — APPROVED** [review]

**Pull request merged**